src/containers/DonorList/index.js

```diff
@@ -28,10 +33,42 @@ export class DonorList extends Component {
         })
     }
 
+
+    handleSubmit = (e, id, index) => {
+       e.preventDefault();
+        const { deleteDonor } = this.props;
+        deleteDonor(id).then(response => {
+            this.setState({ submitted: true })
+            if (response.type === 'SUCCESS') {
+                this.setState({
+                    success: true,
+                    data: this.state.data.filter((_, i) => i !== index)})
+            }
+            if (response.type === 'FAILURE') {
+                this.setState({ success: false })
+            }
+            setTimeout(() => {
+                this.setState({ submitted: false });
```

Comment:
You could probably add `this.props.history.push('/donors')` here so that on successful creation, the page gets redirected to the donors page being consistent with the usual web app behavior. But I think you probably should also check if `success=true` before doing that. If the edit failed, we'd probably want the user to stay on the same page.